### PR TITLE
feat(nwaku): add protected topics

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

**Review by gabrielmer — APPROVED** [review]
LGTM, thank you!

File: wakuv2/nwaku.go [hunk 1]
```diff
@@ -1001,17 +1003,24 @@ func (w *Waku) subscribeToPubsubTopicWithWakuRelay(topic string, pubkey *ecdsa.P
...
+	}
+
+	if len(rs) == 0 {
+		w.logger.Warn("could not obtain shards from topic", zap.String("topic", topic))
+		return nil
+	}
 
 	if pubkey != nil {
-		err := w.node.Relay().AddSignedTopicValidator(topic, pubkey)
+		err := w.node.RelayAddProtectedShard(rs[0].ClusterID, rs[0].ShardIDs[0], pubkey)
```
Comment (Ivansete-status) [inline wakuv2/nwaku.go:1017]: I wonder if we should also make sure `rs[0].ShardIDs.len != 0`

File: wakuv2/nwaku.go [hunk 2]
```diff
@@ -2642,6 +2660,37 @@ func (n *WakuNode) RelaySubscribe(pubsubTopic string) error {
 	return errors.New(errMsg)
 }
 
+func (n *WakuNode) RelayAddProtectedShard(clusterId uint16, shardId uint16, pubkey *ecdsa.PublicKey) error {
+	if pubkey == nil {
+		return nil // Nothing to do here
```
Comment (Ivansete-status) [inline wakuv2/nwaku.go:2665]: Isn't it better to return an error in this case?

**Review by Ivansete-status — COMMENTED** [review]
Sorry for delay. I'm just adding a couple of nitpick comment to maybe consider in the future.

**Pull request merged**